examples/MyModule/recording_backend_soundclick.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ *  recording_backend_soundclick.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Includes from nestkernel
+#include "recording_device.h"
+
+// Own includes
+#include "recording_backend_soundclick.h"
+
+
+// Constructor
+nest::RecordingBackendSoundClick::RecordingBackendSoundClick()
+{
+#ifdef HAVE_SFML_AUDIO
+  // Load the raw sound data into the SFML sound buffer.
+  sound_buffer_.loadFromMemory(
+    sound_click_16bit_44_1khz_wav, sizeof( sound_click_16bit_44_1khz_wav ) );
+  sound_.setBuffer( sound_buffer_ );
+#endif
+}
+
+// Destructor
+nest::RecordingBackendSoundClick::~RecordingBackendSoundClick() throw()
+{
+  finalize();
+}
+
+// Called by each spike detector during network calibration before
+// simulation run
+void
+nest::RecordingBackendSoundClick::enroll( const RecordingDevice& device )
+{
+  // Start or resume, repectively, the stopwatch, which represents the
+  // real time.
+  stopwatch_.start();
+}
```

Comment:
This should definitely throw an exception in case `HAVE_SFML_AUDIO` is undefined. As it is now, the user won't even be notified that no recordings will be made and the setting of the recording backend failed.